### PR TITLE
added setBackground attribute in xml

### LINKED ISSUE #23: Option to set a background color for the entire barview
Include an attribute in the xml tag of the library to directly specify a background colour. Alternatively, define a `setBackgroundColor(Color color)` function in the library to allow dynamic modification of the background color.

### PR DESCRIPTION
**Pull request by BhuwanChandra**
feat: Option to set a background color for the entire barview

added option to set background color or gradient for entire barview

**to set background color:**
```
<in.krharsh17.barview.BarView
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:id="@+id/barview"
        app:backgroundColor="#f7f7f7"
        />
```

**to set background gradient:**
```
<in.krharsh17.barview.BarView
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:id="@+id/barview"
        app:gradientStart="#FAA5A5"
        app:gradientEnd="#AEBCFB"
        />
```

Resolves #23

### CODE REVIEW

File: library/src/main/java/in/krharsh17/barview/BarView.java [hunk 1]
```diff
@@ -130,6 +136,17 @@ private void populateBarView() {
         }
     }
 
+    public void setBackgroundColor(String color){
+        containerLayout.setBackgroundColor(Color.parseColor(color));
+    }
+
+    public void setBackgroundGradient(String startColor, String endColor, String direction){
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarView.java]: What's the use of `direction` here?

File: library/src/main/java/in/krharsh17/barview/BarView.java [hunk 2]
```diff
@@ -138,6 +155,7 @@ private void addBar(BarModel data) {
                 data.getValue(),
                 data.getFillRatio(),
             barMargin,
+
```
Comment (krharsh17) [inline library/src/main/java/in/krharsh17/barview/BarView.java]: Please remove this

File: library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java [hunk 3]
```diff
@@ -8,6 +8,8 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import static org.junit.Assert.assertEquals;
+
```
Comment (krharsh17) [inline library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java:12]: Is this necessary?

**Review by krharsh17 — CHANGES_REQUESTED** [review]

File: library/src/main/java/in/krharsh17/barview/BarView.java (hunk 1 above)
Comment (BhuwanChandra) [reply library/src/main/java/in/krharsh17/barview/BarView.java]: it's for the direction of the gradient(left to right, top to bottom). currently, it is left to right. 
just want to ask, do we need to add options for direction for gradient or not?

File: library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java (hunk 3 above)
Comment (BhuwanChandra) [reply library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java:12]: you just used assertEquals and didn't imported it, so, it was showing error. that's why I imported it

File: library/src/main/java/in/krharsh17/barview/BarView.java (hunk 2 above)
Comment (BhuwanChandra) [reply library/src/main/java/in/krharsh17/barview/BarView.java]: ok

File: library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java (hunk 3 above)
Comment (krharsh17) [reply library/src/androidTest/java/in/krharsh17/barview/ExampleInstrumentedTest.java:12]: Alright

File: library/src/main/java/in/krharsh17/barview/BarView.java (hunk 1 above)
Comment (krharsh17) [reply library/src/main/java/in/krharsh17/barview/BarView.java]: Yes, that would be preferable. A `bg_gradient_direction` attribute for the barview, with two options: vertical & horizontal

**Comment by BhuwanChandra** [comment]
gradient direction added 

**to set gradient background :** 
<in.krharsh17.barview.BarView
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:id="@+id/barview"
        app:gradientStart="#FAA5A5"
        app:gradientEnd="#AEBCFB"
        app:gradientDirection="vertical"
        />

**values:** 
"vertical", "horizontal"

**default value:**
 "horizontal"

**Review by samagragupta — APPROVED** [review]

**Review by samagragupta — CHANGES_REQUESTED** [review]
Resolve conflicts

**Comment by saumya1singh** [comment]
Conflicts should be resolved before merging. Please check @BhuwanChandra !

**Comment by krharsh17** [comment]
There's a quality issue with a switch statement. @BhuwanChandra Please fix that and it will be merged.

**Review by krharsh17 — APPROVED** [review]

**Pull request merged**